src/pages/instances/forms/InstanceLocationSelect.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { FC, useState } from "react";
+import { Select } from "@canonical/react-components";
+import { useQuery } from "@tanstack/react-query";
+import { queryKeys } from "util/queryKeys";
+import Loader from "components/Loader";
+import { useSettings } from "context/useSettings";
+import { fetchClusterGroups, fetchClusterMembers } from "api/cluster";
+import { FormikProps } from "formik/dist/types";
+import { CreateInstanceFormValues } from "pages/instances/CreateInstanceForm";
+
+interface Props {
+  formik: FormikProps<CreateInstanceFormValues>;
+}
+
+const figureDefaultGroup = (target?: string) => {
+  if (!target?.startsWith("@")) {
+    return "default";
+  }
+  return target.split("@")[1];
+};
+
+const figureDefaultMember = (target?: string) => {
+  if (!target || target.startsWith("@")) {
+    return "";
+  }
+  return target;
+};
+
+const InstanceLocationSelect: FC<Props> = ({ formik }) => {
+  const defaultGroup = figureDefaultGroup(formik.values.target);
+  const [selectedGroup, setSelectedGroup] = useState(defaultGroup);
+  const defaultMember = figureDefaultMember(formik.values.target);
+  const [selectedMember, setSelectedMember] = useState(defaultMember);
+
+  const { data: settings } = useSettings();
+  const isClustered = settings?.environment?.server_clustered;
+
+  if (!isClustered) {
+    return <></>;
+  }
+
+  const { data: members = [], isLoading: isMembersLoading } = useQuery({
+    queryKey: [queryKeys.cluster, queryKeys.members],
+    queryFn: fetchClusterMembers,
+    enabled: isClustered,
+  });
+
+  const { data: clusterGroups = [], isLoading: isGroupsLoading } = useQuery({
+    queryKey: [queryKeys.cluster, queryKeys.groups],
+    queryFn: fetchClusterGroups,
+  });
+
+  if (isMembersLoading || isGroupsLoading) {
+    return <Loader />;
+  }
+
+  const setGroup = (group: string) => {
+    formik.setFieldValue("target", `@${group}`);
+    setSelectedGroup(group);
+    setSelectedMember("");
+  };
+
+  const setMember = (member: string) => {
+    if (member === "") {
+      setGroup(selectedGroup);
+    } else {
+      formik.setFieldValue("target", member);
+      setSelectedMember(member);
+    }
+  };
+
+  return (
+    <>
+      <Select
+        id="locationGroup"
+        label="Location group"
+        onChange={(e) => setGroup(e.target.value)}
+        value={selectedGroup}
+        options={clusterGroups.map((group) => {
+          return { label: group.name, value: group.name };
+        })}
+      />
+      <Select
+        id="locationMember"
+        label="Location member"
+        onChange={(e) => setMember(e.target.value)}
+        value={selectedMember}
+        options={[
+          { label: "any", value: "" },
```

Comment:
Maybe we should not show the "any" option if there are 0 members or 1 member. I did not remember that the group I was selecting had zero members and left "any" selected, and got the "No suitable cluster member could be found" error after trying to create the instance - a tedious postponed feedback. We could show a hint below explaining what the problem is and disable the create buttons.
Even better, we could also filter the `clusterGroup` list above so that we don't include groups that have zero members.

Reply:
I think we should disable the empty groups, so they can't be selected.